Polls: Show who votes for each option

rsashank opened the pull request:
<!-- See README for documentation, or ask in #zulip-terminal if unclear -->
### What does this PR do, and why?
Add support to view polls

### External discussion & connections
<!-- [x] all that apply, specifying topic and adding numbers after # for issues/PRs -->
- [x] Discussed in **#zulip-terminal** in `Polls - Widget Support #T986 #T1477`
- [ ] Fully fixes #
- [x] Partially fixes issue #986 
- [x] Builds upon previous unmerged work in PR #1549 (first 2 commits)
- [ ] Is a follow-up to work in PR #
- [ ] Requires merge of PR #
- [ ] Merge will enable work on #

### How did you test this?
<!-- [x] all that apply -->
- [x] Manually - Behavioral changes
- [x] Manually - Visual changes
- [x] Adapting existing automated tests
- [x] Adding automated tests for new behavior (or missing tests)
- [x] Existing automated tests should already cover this (*only a refactor of tested code*)

### Self-review checklist for each commit
- [x] It is a [minimal coherent idea](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development)
- [x] It has a commit summary following the [documented style](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development) (title & body)
- [x] It has a commit summary describing the  motivation and reasoning for the change
- [x] It individually passes linting and tests
- [x] It contains test additions for any new behavior
- [x] It flows clearly from a previous branch commit, and/or prepares for the next commit

### Visual changes    <!-- DELETE SECTION IF NO VISUAL CHANGE -->
| Before | After | Webapp |
|--------|--------|--------|
| ![image](https://github.com/user-attachments/assets/c2736167-7bb6-40ec-9fcd-f2100c2687be)| ![image](https://github.com/user-attachments/assets/4c117c32-92e7-4b68-9a59-ba64494f9e52)| ![image](https://github.com/user-attachments/assets/93268e30-1edc-4ec7-bef3-4719721f0394) |

rsashank committed:
keys: Add SHOW_POLL_VOTES hotkey.

SHOW_POLL_VOTES : 'v' to check votes for a given poll.

rsashank committed:
views: Add PollResultsView class.

Takes poll_question and options as an input and displays
voters for each option.

rsashank committed:
core: Add show_poll_vote method to Controller.

Creates an instance of PollResultsView class and shows the popup.

rsashank committed:
messages: Add logic to handle the SHOW_POLL_VOTES hotkey.

neiljp reviewed (commented):
@rsashank This update looks good, and being able to see the poll votes is useful. This looks fine in a separate popup for now :+1:

Having polls update via events would be the obvious next step here, though we may want to focus on merging the todo PR first to make integrating this easier?

neiljp reviewed (commented):
@rsashank Thanks for the update - this looks rather close to being ready 🎉 I tested this manually and had to remind myself why the poll updates were working even though you didn't add them in this PR! 😮 

Generally, there are a few nits I noticed similar to the other PR and to avoid duplication, a few improvements to the popup, and some small possible restructuring.

The main point left to discuss is probably the key to use - or where to connect the popup to. That's not *critical* before merging, though I'm often wary of accepting a new hotkey unless we know it's "safe" and won't impact things later. It's easy enough to change in a later commit, but since we're not currently releasing regularly then in principle anyone may use this and start using the hotkey!

If the latter becomes a sticking point, we can certainly merge the main part first, and leave the popup part to another PR (or similar).

rsashank commented on zulipterminal/core.py:
```diff
@@ -281,6 +282,36 @@ def show_msg_info(
+            for voter_id in voter_ids:
+                user_info = self.model.get_user_info(voter_id)
+                if user_info:
+                    voter_names.append(user_info["full_name"])
```
Oops my bad, I had a response written that I forgot to submit as a review.

neiljp committed:
WIP: keys: Adjust poll votes popup to 'M'.

Using 'w' clashes with the global search-users hotkey.

Use of 'M' is only temporary, to enable feature demonstration.

neiljp commented:
@rsashank I've merged the first two commits and pushed back here, both with minor adjustments.

I'm marking this as a completion candidate for now, since there were a few outstanding concerns from what I could tell. #1573 is intended to track further poll widget work.

neiljp commented:
Updated PR name to reflect remaining commits.
